infra/base-images/base-runner/gocoverage/llvm-cov-rel/llvm-profraw-relative.go

```diff
@@ -0,0 +1,180 @@
+	value := binary.LittleEndian.Uint64(data[offset : offset+8])
+	// if the value is an address in the right section,
+	if value >= sectPrfCnts && value < sectPrfData {
+		// make it relative
+		value = value - databegin
+		binary.LittleEndian.PutUint64(data[offset:offset+8], value)
+	}
+
+}
+
+func profrawDataLen(ipvklast uint64) int {
+	return 44 + 2*(int(ipvklast)+1)
+}
+
+func relativizeProfraw(data []byte, sectPrfCnts uint64, sectPrfData uint64) (error, []byte) {
+	h := ProfrawHeaderVersion7{}
+	var err error
+	h.ProfrawHeaderGeneric, err = parseProfrawHeaderGeneric(data)
+	if err != nil {
+		return err, data
+	}
+	if h.Version == 5 {
+		// upgrade from 5 to 7 by adding a zero binaryids in the header
+		binary.LittleEndian.PutUint64(data[8:16], 7)
+		h.Version = 7
+		data2 := make([]byte, len(data)+8)
+		copy(data2, data[0:16])
+		copy(data2[24:], data[16:])
+		data = data2
+	}
+	if h.Version != 7 {
+		// at one point clang-14 will update to 8, and so should we ()
+		return fmt.Errorf("Invalid version for profraw file: %v", h.Version), data
+	}
+	if len(data) < PROFRAW_HEADER_7_LEN {
+		return io.EOF, data
+	}
+	h.BinaryIdsSize = binary.LittleEndian.Uint64(data[16:24])
+	h.DataSize = binary.LittleEndian.Uint64(data[24:32])
+	h.PaddingBytesBeforeCounters = binary.LittleEndian.Uint64(data[32:40])
+	h.CountersSize = binary.LittleEndian.Uint64(data[40:48])
+	h.PaddingBytesAfterCounters = binary.LittleEndian.Uint64(data[48:56])
+	h.NamesSize = binary.LittleEndian.Uint64(data[56:64])
+	h.CountersDelta = binary.LittleEndian.Uint64(data[64:72])
+	h.NamesDelta = binary.LittleEndian.Uint64(data[72:80])
+	h.ValueKindLast = binary.LittleEndian.Uint64(data[80:88])
+
+	if h.CountersDelta != sectPrfCnts-sectPrfData {
+		// Rust linking adds an offset ? not seen in readelf
+		sectPrfData = h.CountersDelta - sectPrfCnts + sectPrfData
+		sectPrfCnts = h.CountersDelta
+	}
+	dataref := sectPrfData
+	relativizeAddress(data, 64, dataref, sectPrfCnts, sectPrfData)
+
+	offset := PROFRAW_HEADER_7_LEN + int(h.BinaryIdsSize)
+	for i := uint64(0); i < h.DataSize; i++ {
+		if len(data) < offset+profrawDataLen(h.ValueKindLast) {
+			return io.EOF, data
+		}
+		d := ProfrawData{}
+		d.NameRef = binary.LittleEndian.Uint64(data[offset : offset+8])
+		d.FuncHash = binary.LittleEndian.Uint64(data[offset+8 : offset+16])
+		d.CounterPtr = binary.LittleEndian.Uint64(data[offset+16 : offset+24])
+		d.FunctionPointer = binary.LittleEndian.Uint64(data[offset+24 : offset+32])
+		d.Values = binary.LittleEndian.Uint64(data[offset+32 : offset+40])
+		d.NumCounters = binary.LittleEndian.Uint32(data[offset+40 : offset+44])
+		d.NumValueSites = make([]uint16, h.ValueKindLast+1)
+		for j := 0; j <= int(h.ValueKindLast); j++ {
+			d.NumValueSites[j] = binary.LittleEndian.Uint16(data[offset+44+2*j : offset+46+2*j])
+		}
+
+		relativizeAddress(data, offset+16, dataref, sectPrfCnts, sectPrfData)
+		// cf CountersDelta -= sizeof(*SrcData); in __llvm_profile_merge_from_buffer
```

Comment:
Why not delete this?

Reply:
This comment explains the addition afterwards